src/App.tsx

```diff
@@ -1,39 +1,63 @@
-import React from 'react';
+import { PureComponent } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+type State = {
+  clockName: string,
+  hasClock: boolean,
+};
 
-  return `Clock-${value}`;
-}
+export class App extends PureComponent<{}, State> {
+  state: Readonly<State> = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerIdName = 0;
+
+  value = Date.now().toString().slice(-4);
+
+  componentDidMount() {
+    document.addEventListener('click', () => {
+      this.setState({ hasClock: true });
+    });
```

Comment:
Same here: Create separate handler. Do not forget to remove listeners on component unmount